can_sam.c fails to compile when CONFIG_NETWORKING is set

**Describe the bug**
Attempting to build an application with both CONFIG_CAN and CONFIG_NETWORKING enabled produces a compilation error.

**To Reproduce**
`west build -b sam_e70_xplained tests/drivers/can/api/ -- -DCONFIG_NETWORKING=y
`
**Expected behavior**
The test harness should build without error.

**Impact**
Currently unable to add CAN functionality to my application which also uses networking.

**Logs and console output**
```
drivers/can/can_sam.c:169:44: error: section of 'can_sam_mram_0' conflicts with previous declaration
  169 |         CAN_MCAN_DT_INST_MRAM_DEFINE(inst, can_sam_mram_##inst);        \
      |                                            ^~~~~~~~~~~~~
```

**Environment (please complete the following information):**
 - OS: Linux 6.3.12-200.fc38.x86_64
 - Toolchain: Zephyr SDK 0.16.3
 - Version: zephyr-v3.5.0

**Additional context**

--- a/include/zephyr/drivers/can/can_mcan.h
+++ b/include/zephyr/drivers/can/can_mcan.h
@@ -640,7 +640,7 @@ enum can_mcan_psr_lec {
  */
 #define CAN_MCAN_DT_MRAM_DEFINE(node_id, _name)                                                    \
 	BUILD_ASSERT(CAN_MCAN_DT_MRAM_OFFSET(node_id) == 0, "offset must be 0");                   \
-	static char __noinit __nocache __aligned(4) _name[CAN_MCAN_DT_MRAM_ELEMENTS_SIZE(node_id)];
+	static char __nocache_noinit __aligned(4) _name[CAN_MCAN_DT_MRAM_ELEMENTS_SIZE(node_id)];
 
 /**
  * @brief Assert that the Message RAM configuration meets the Bosch M_CAN IP core restrictions

--- a/include/zephyr/linker/section_tags.h
+++ b/include/zephyr/linker/section_tags.h
@@ -49,8 +49,10 @@
 
 #if defined(CONFIG_NOCACHE_MEMORY)
 #define __nocache __in_section_unique(_NOCACHE_SECTION_NAME)
+#define __nocache_noinit __nocache
 #else
 #define __nocache
+#define __nocache_noinit __noinit
 #endif /* CONFIG_NOCACHE_MEMORY */
 
 #if defined(CONFIG_KERNEL_COHERENCE)

